### DISCUSSION TITLE
STM32F103C8T6: Unsupported architecture

### DISCUSSION THREAD

**Discussion by TheGoldArmada** [ZZ-Cat/CRSFforArduino#122] [Q&A] [answered]
Hi, I came across your library, it looks very cool, but from the entire list of available architectures, I only have my STM32, but as you know, there is no choice of this board in the Arduino IDE by default. I found the Arduino_STM32-master library, successfully installed and tested it, everything works. But when I try to run your sketch "link_stats" and select the "Generic STM32F103C series" board, I get the "Unsupported architecture" error. Are there any ways to solve this problem, such as using other specific libraries for STM32?

**Comment by TheGoldArmada** [comment]
Also, by default, I get the error "cstring: No such file or directory" in the file SerialBuffer.cpp but after I change cstring to string.h, this error disappears. Can such a replacement interfere with correct operation?

**Comment by ZZ-Cat** [comment, marked as answer]
Can you re-submit this as an issue, please?

Also, I had to cherry-pick some of the STM32F103 targets, because some of them didn't have the flash memory space available for all possible configurations of CRSF for Arduino. Either that or when I _did_ fully configure the library, there was something along the lines of 60~70 % of flash memory used by the library alone, and I felt that wasn't realistic for me to include _that_ particular variant of the STM32F103 target.

I understand that STM32F103 targets have, by and large, been phased out in the context of projects such as Betaflight, RotorFlight, and INAV for similar reasons why I had to cherry-pick supporting some STM32F103 targets and not others.

Simply put, all STM32F103 targets have _very small_ memory footprints.  
It _could_ be done. However, in the near future, I plan to mark support of the entore STM32F103 list as "End of Life", because as my project grows, these targets MAY NOT have the memory space available to support CRSF for Arduino anymore.
